src/protocol/gossip/gossip.cpp

```diff
@@ -729,8 +730,34 @@ namespace libp2p::protocol::gossip {
           add_peer(peer);
         }
       }
-      for (auto &peer : topic.mesh_peers_) {
-        add_peer(peer);
+      if (config_.soon) {
+        auto mesh_n = config_.mesh_n_for_topic(topic.topic_hash_);
+        if (topic.mesh_peers_.size() <= mesh_n) {
```

Comment:
[Research mentions new config field `Δ ≤ mesh_n`](https://ethresear.ch/t/selecting-optimal-outbound-neighbors-soon-for-fast-bandwidth-efficient-propagation-in-p2p-networks/23358).

Gossip tries to graft `mesh_n` peers when there are less than `mesh_n_low` mesh peers.
Mesh peers may grow up to `mesh_n_high` when other peers graft us, but it's not guaranteed.

Reply:
Updated accordingly: 
* replaced `bool soon = false` with `int soon_delta = 1` defining top number of peers by RTT from the mesh to which peer propagates messages